rasa/shared/core/training_data/visualization.py

```diff
@@ -454,15 +478,44 @@ def visualize_neighborhood(
                 idx -= 1
                 break
             if isinstance(el, UserUttered):
+                if message is not None:
+                    logger.error("multiple messages or slots in a row!!")
```

Comment:
This still needs to be addressed with indication on how to resolve the error. Also not sure if logging is appropriate here if the error lies with the user sending multiple messages?

Reply:
As visualize is taken from the training dataset, i don't think there is such a case in the events where a UserUttered is multiple messages in a row. regarding multiple slots, it would be taken from SlotSet event and not UserUttered. This check is not needed imo @ancalita

Reply:
It can only happen if you have an error in the training stories, like two consecutive intents.

Reply:
When I developed the code last year, I also tried to break it. That was one way to make the python fail or make a nonsense diagram without any warning, so I added the (admittedly very short) error message here.